[CodeGen] Allow multiple location for the same CSR.

mgudim opened the pull request:
Currently there is an implicit assumption in `CFIInstrInserter` that a CSR can be saved only in one location. However, scenarios when this assumption breaks are possible.

Commit:
[CodeGen] Allow multiple location for the same CSR.

Currently there is an implicit assumption in `CFIInstrInserter` that a CSR can
be saved only in one location. However, scenarios when this assumption breaks are possible.

ppenzin commented on llvm/lib/CodeGen/CFIInstrInserter.cpp:
```diff
@@ -76,38 +118,27 @@ class CFIInstrInserter : public MachineFunctionPass {
+    SmallVector<CSRSavedLocation> IncomingCSRLocations;
+    /// Set of locations where the callee saved registers are at basic block
+    /// exit.
+    SmallVector<CSRSavedLocation> OutgoingCSRLocations;
```
Are all fields inside the `CSRSavedLocation` needed downstream from this definition? These two data structures would use quite a bit more memory, especially if we have a target with a lot of registers.

mgudim committed:
[CFIInstrInserter] Only track callee-saved registers.

Since we only emit CFIs for callee-saved registers, we should track only
them. Before this commit `CFIInstrInserter` tracked all the registers.

This will save us some memory, which is especially important for
upcoming PRs.
